google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/SpannerPool.java

```diff
@@ -349,6 +352,9 @@ Spanner createSpanner(SpannerPoolKey key, ConnectionOptions options) {
         .setDatabaseRole(options.getDatabaseRole())
         .setCredentials(options.getCredentials());
     builder.setSessionPoolOption(key.sessionPoolOptions);
+    if (options.isEnabledOpenTelemetryMetrics() && options.getOpenTelemetry() != null) {
```

Comment:
`options.getOpenTelemetry()` will in the current implementation never return null, so I think it would be better to remove this null check, and instead add a `NonNull` annotation to the `getOpenTelemetry()` method.

Reply:
Fixed.